react/features/filmstrip/components/native/Filmstrip.js

```diff
@@ -149,32 +180,87 @@ class Filmstrip extends Component<Props> {
     }
 
     /**
-     * Sorts a specific array of {@code Participant}s in display order.
+     * Returns the participant list in a format suitable for the
+     * {@code FlatList} component to display.
+     *
+     * @private
+     * @returns {Array<{participant: Object, key: string}>}
+     */
+    _getFilmstripData() {
+        const { _participants } = this.props;
+        const data = [];
+
+        // If _separateLocalThumbnail is false (e.g. on Android for now, see
+        // comment in constructor for more details) we render the local
+        // participant together with the remote participants, so we need to put
+        // it as the first element of the data array.
+        if (!this._separateLocalThumbnail) {
+            const { _localParticipant } = this.props;
+
+            data.push({
+                participant: _localParticipant,
+                key: _localParticipant.id
+            });
+        }
+
+        if (_participants) {
+            for (const p of _participants) {
+                data.push({
+                    key: p.id,
+                    participant: p
+                });
+            }
+
+            if (isNarrowAspectRatio(this)) {
+                data.reverse();
+            }
+        }
+
+        return data;
+    }
+
+    _onViewableItemsChanged: Object => void;
+
+    /**
+     * Called when the viewability of thumbnails changes, as defined by the
+     * viewabilityConfig prop.
      *
-     * @param {Participant[]} participants - The array of {@code Participant}s
-     * to sort in display order.
-     * @param {boolean} isNarrowAspectRatio_ - Indicates if the aspect ratio is
-     * wide or narrow.
      * @private
-     * @returns {Participant[]} A new array containing the elements of the
-     * specified {@code participants} array sorted in display order.
+     * @param {Object} changeEvent - Info about the item thats visibility has
+     * changed.
+     * @returns {void}
      */
-    _sort(participants, isNarrowAspectRatio_) {
-        // XXX Array.prototype.sort() is not appropriate because (1) it operates
-        // in place and (2) it is not necessarily stable.
-
-        const sortedParticipants = [
-            ...participants
-        ];
-
-        if (isNarrowAspectRatio_) {
-            // When the narrow aspect ratio is used, we want to have the remote
-            // participants from right to left with the newest added/joined to
-            // the leftmost side. The local participant is the leftmost item.
-            sortedParticipants.reverse();
+    _onViewableItemsChanged(changeEvent) {
+        if (changeEvent
+                && changeEvent.viewableItems
+                && changeEvent.viewableItems.length) {
```

Comment:
What is the length check for? Is it possible for it to be 0 and needed to be acted upon?

Reply:
I observed that there are update events sometimes that contains an empty item list (I suppose this is when the list is re-rendered) that should be excluded, otherwise the Filmstrip will start to blink.